src/main/clojure/jarl/builtins/array.clj

```diff
@@ -19,13 +19,25 @@
 
 (defn builtin-concat [a b]
   (when-not (vector? a)
-    (throw (BuiltinException. (format "arg 0 is not an array"))))
+    (throw (BuiltinException. "arg 0 is not an array")))
```

Comment:
I thought about this a bit, and I think we might want to remove these type assertions entirely, or at least not throw an exception by default. I'll assume that when OPA builds the IR represenation, it will type check whatever it can already - i.e. anything "static", so e.g. `array.concat([1, 2, 3], "1", "2")` is going to fail at that stage already.

The case that remains then is only when the args are unknown at compile time, like when they're provided as part of the input (or some `data` attribute). In those cases however, OPA does not error by default, but returns undefined.

```shell
$ echo '{"x": "1"}' | opa eval --stdin-input 'array.concat([1], input.x)'
{}
```
This may of course be tweaked by using `--strict-builtin-errors`:
```shell
echo '{"x": "1"}' | opa eval --stdin-input --strict-builtin-errors 'array.concat([1], input.x)'
{
  "errors": [
    {
      "message": "array.concat: operand 2 must be array but got string",
      "code": "eval_type_error",
      "location": {
        "file": "",
        "row": 1,
        "col": 1
      }
    }
  ]
}
```
We might want to have that same option, but I think we should use the same defaults as OPA here. I agree that we should try and mimic the error messages of OPA too, so I'll update the error messages in the meantime.

On a tangentially related note, I wonder if there's a way for us to get the location of the error from the IR?

Reply:
I've made an update to better mirror the OPA error messages, but above concern remains. We can tackle that later though :)

Reply:
There definitely is a way to get the locations in there, as it's part of the IR. Have just been ignoring that info thus far.

I think you're right also about the fact that all the args should already be aligned by the IR. My thinking here was to go with strict errors until we add that option, at which point we'll make `undefined` results the default.

Reply:
Great! Let's create an issue for including location too 👍